app/controllers/concerns/neighbourhood_filter.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+class NeighbourhoodFilter
+  attr_reader :neighbourhoods
+
+  def initialize(current_site, neighbourhoods, params)
+    @current_neighbourhood = Neighbourhood.find_by(id: params[:neighbourhood])
+    @neighbourhoods = neighbourhoods
+    @badge_zoom_level = current_site.badge_zoom_level
+  end
+
+  def active?
+    @current_neighbourhood.present?
+  end
+
+  def current_neighbourhood?(neighbourhood)
+    @current_neighbourhood.present? && (@current_neighbourhood.id == neighbourhood.id)
+  end
+
+  def apply_to_partner(query = Partner)
```

Comment:
a little bit of a confusing name as it is not being applied to a partner but applying the filter to the query chain